System/Box.h

```diff
@@ -1,516 +1,294 @@
+		/// </summary>
+		/// <param name="newCorner">A Vector with the new primary corner.</param>
+		void SetCorner(const Vector &newCorner) { m_Corner = newCorner; }
+
+		/// <summary>
+		/// Gets the center point of this Box' area, in absolute Scene coordinates.
+		/// </summary>
+		/// <returns>The center point.</returns>
+		Vector GetCenter() const { return Vector(m_Corner.m_X + (m_Width / 2), m_Corner.m_Y + (m_Height / 2)); }
+
+		/// <summary>
+		/// Sets the primary corner of this box, by specifying where the center ought to be.
+		/// </summary>
+		/// <param name="newCenter">A Vector with the new center point.</param>
+		void SetCenter(const Vector &newCenter) { m_Corner.SetXY(newCenter.m_X - (m_Width / 2), newCenter.m_Y - (m_Height / 2)); }
+
+		/// <summary>
+		/// Gets the width of this box. Note that this can be negative if the box hasn't been righted with Unflip().
+		/// </summary>
+		/// <returns>A float value that represents the width value of this Box.</returns>
+		float GetWidth() const { return m_Width; }
+
+		/// <summary>
+		/// Sets the width of this box. Note that this can be negative if the box hasn't been righted with Unflip().
+		/// </summary>
+		/// <param name="width">A float value that represents the width value of this Box.</param>
+		void SetWidth(float width) { m_Width = width; }
+
+		/// <summary>
+		/// Gets the height of this box. Note that this can be negative if the box hasn't been righted with Unflip().
+		/// </summary>
+		/// <returns>A float value that represents the height value of this Box.</returns>
+		float GetHeight() const { return m_Height; }
+
+		/// <summary>
+		/// Sets the height of this box. Note that this can be negative if the box hasn't been righted with Unflip().
+		/// </summary>
+		/// <param name="height">A float value that represents the height value of this Box.</param>
+		void SetHeight(float height) { m_Height = height; }
+
+		/// <summary>
+		/// Gets the width times the height.
+		/// </summary>
+		/// <returns>The width times the height.</returns>
+		float GetArea() const { return m_Width * m_Height; }
+
+		/// <summary>
+		/// Gets a random point within this box.
+		/// </summary>
+		/// <returns>The random point within the box.</returns>
+		Vector GetRandomPoint() const { return Vector(m_Corner.m_X + m_Width * PosRand(), m_Corner.m_Y + m_Height * PosRand()); }
+#pragma endregion
+
+#pragma region Detection
+		/// <summary>
+		/// Tells whether another box intersects this one.
+		/// </summary>
+		/// <param name="rhs">The other Box to check for intersection with.</param>
+		/// <returns>Intersecting the other box or not.</returns>
+		bool IntersectsBox(const Box &rhs);
+
+		/// <summary>
+		/// Tells whether a point is within the Box or not.
+		/// </summary>
+		/// <param name="point">The Vector describing the point to test for within box bounds.</param>
+		/// <returns>Inside the box or not.</returns>
+		bool WithinBox(const Vector &point) const;
```

Comment:
I'm not sure about renaming this, at least at this point.
First, it's gonna drag a ton of changes in other files.
Second, gonna break everything lua that uses it.

Reply:
Yeah if we rename it we should rename the binding too. It will mean changing a bunch of files (though refactor -> rename can help with that) so sure, we can do it in another smaller piece of work somewhere that can get quickly done and merged. If you wanna do it that way please put in an issue.